src/util/list.c

```diff
@@ -118,18 +123,32 @@ int append_list(list_t *list, void *element)
 
 void for_each_list(list_t *list, listIterator iterator)
 {
-    assert(iterator != NULL);
 
     list_node_t *node = list->head;
-    bool result = true;
+    BOOL result = TRUE;
```

Comment:
Since bool is now part of the standard library, isn't it preferable to use it as opposed to our custom implementation?